source/extensions/filters/http/gcp_authn/gcp_authn_filter.h

```diff
@@ -34,6 +38,58 @@ struct GcpAuthnFilterStats {
 using FilterConfigProtoSharedPtr =
     std::shared_ptr<envoy::extensions::filters::http::gcp_authn::v3::GcpAuthnFilterConfig>;
 
+template <typename TokenType>
+using LRUCache = ::google::simple_lru_cache::SimpleLRUCache<std::string, TokenType>;
+using JwtToken = ::google::jwt_verify::Jwt;
+
+template <typename TokenType> class TokenCacheImpl : public Logger::Loggable<Logger::Id::init> {
+public:
+  TokenCacheImpl(const envoy::extensions::filters::http::gcp_authn::v3::TokenCacheConfig& config,
+                 TimeSource& time_source)
+      : time_source_(time_source) {
+    lru_cache_ = std::make_unique<LRUCache<TokenType>>(config.cache_size());
+  }
+
+  TokenCacheImpl() = delete;
+  TokenType* lookUp(const std::string& key);
+  void insert(const std::string& key, std::unique_ptr<TokenType>&& token);
+
+  LRUCache<TokenType>& lruCache() { return *lru_cache_; }
+
+  ~TokenCacheImpl() {
+    ASSERT(lru_cache_ != nullptr);
+    // Remove all entries from the cache.
+    lru_cache_->clear();
+  }
+
+private:
+  std::unique_ptr<LRUCache<TokenType>> lru_cache_;
```

Comment:
Why do you need to own it through a `std::unique_ptr`. Why no own `LRUCache<TokenType>` directly.

Reply:
Good point. `unique_ptr` doesn't seem to be necessary.
Will update to use bare object